Incorrect suggestion for method call on ambiguous floating-point literal

[Given the following code:](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=88c7a77f5b6fc63b5d1f18620933b1e3)

```rust
println!("{}", (3.).recip());
```

The current output is:

```
error[E0689]: can't call method `recip` on ambiguous numeric type `{float}`
 --> src/main.rs:2:25
  |
2 |     println!("{}", (3.).recip());
  |                         ^^^^^
  |
help: you must specify a concrete type for this numeric value, like `f32`
  |
2 |     println!("{}", (3._f32).recip());
  |                     ~~~~~~

For more information about this error, try `rustc --explain E0689`.
```

However, this won't compile, since rustc doesn't accept `3._f32` as a floating-point literal. The `help` should add the implied `0` for us:

```
2 |     println!("{}", (3.0_f32).recip());
  |                     ~~~~~~~
```

See also https://github.com/rust-lang/rust/issues/51874, a similar issue which was fixed.

--- a/compiler/rustc_typeck/src/check/method/suggest.rs
+++ b/compiler/rustc_typeck/src/check/method/suggest.rs
@@ -317,6 +317,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     .span_to_snippet(lit.span)
                                     .unwrap_or_else(|_| "<numeric literal>".to_owned());
 
+                                // If this is a floating point literal that ends with '.',
+                                // get rid of it to stop this from becoming a member access.
+                                let snippet = snippet.strip_suffix('.').unwrap_or(&snippet);
+
                                 err.span_suggestion(
                                     lit.span,
                                     &format!(
@@ -324,7 +328,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                          like `{}`",
                                         concrete_type
                                     ),
-                                    format!("{}_{}", snippet, concrete_type),
+                                    format!("{snippet}_{concrete_type}"),
                                     Applicability::MaybeIncorrect,
                                 );
                             }

--- a/src/test/ui/suggestions/issue-90974.rs
+++ b/src/test/ui/suggestions/issue-90974.rs
@@ -1,0 +1,3 @@
+fn main() {
+    println!("{}", (3.).recip()); //~ERROR
+}

--- a/src/test/ui/suggestions/issue-90974.stderr
+++ b/src/test/ui/suggestions/issue-90974.stderr
@@ -1,0 +1,14 @@
+error[E0689]: can't call method `recip` on ambiguous numeric type `{float}`
+  --> $DIR/issue-90974.rs:2:25
+   |
+LL |     println!("{}", (3.).recip());
+   |                         ^^^^^
+   |
+help: you must specify a concrete type for this numeric value, like `f32`
+   |
+LL |     println!("{}", (3_f32).recip());
+   |                     ~~~~~
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0689`.

